src/components/KTreeList/KTreeList.vue

```diff
@@ -111,6 +111,14 @@ const emit = defineEmits<{
   (event: 'selected', item: TreeListItem): void
 }>()
 
+const { useDebounce } = useUtilities()
+
+// use debounce function to avoid emitting multiple same events at once
+// even a 1ms debounce will do the trick
+const { debouncedFn: debouncedEmit } = useDebounce((item: TreeListItem) => {
+  emit('selected', item)
+}, 1)
```

Comment:
tbh this seems a bit like a hack. Can you explain the actual issue here?

Reply:
I'm unsure what the cause of the problem is, this solution worked for me while I was working on something. I do agree that this might not be ideal solution, I'll move this back to draft and investigate this a bit more when I have time.

Reply:
I think investigating why a 1ms debounce is impacting selection is the way to go.

Is the DOM not yet updated for something? Then you could `await nextTick()` for a DOM cycle instead, etc.